File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/own-tests-project/src/surveycrawler/turn_clockwise_command.py
from surveycrawler.crawler_command import CrawlerCommand


class TurnClockwiseCommand(CrawlerCommand):

    # command character

    @classmethod
    def command_character(cls):
        return 'h'

    # executing

    def execute(self, a_crawler):
        a_crawler.turn_clockwise()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/analysis/own-tests-project/src/main/java/surveycrawler/TurnClockwiseCommand.java
package surveycrawler;

public class TurnClockwiseCommand extends CrawlerMovement {

    // testing

    @Override
    public boolean isFor(char aCharacter) {
        return aCharacter == 'h';
    }

    // executing

    @Override
    public void executeOn(SurveyCrawler aCrawler) {
        aCrawler.turnClockwise();
    }

    // undoing

    @Override
    public CrawlerMovement inverse() {
        return new TurnCounterClockwiseCommand();
    }
}
